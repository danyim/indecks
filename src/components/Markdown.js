import React from 'react';
import Remarkable from 'remarkable';

class Markdown extends React.Component {
  convertToMarkdown(plaintext = '', removeParaTags = false) {
    const md = new Remarkable();
    let rawMarkup = md.render(plaintext);
    if(removeParaTags === true) {
      rawMarkup = rawMarkup.replace('<p>', '');
      rawMarkup = rawMarkup.replace('</p>', '');
    }
    return { __html: rawMarkup };
  }

  render() {
    return (
      <span className={this.props.className}
        dangerouslySetInnerHTML={this.convertToMarkdown(this.props.text, this.props.removeParaTags)} />
    );
  }
}

Markdown.defaultProps = {
  className: '',
  removeParaTags: false,
  text: '',
};
Markdown.propTypes = {
  className: React.PropTypes.string,
  removeParaTags: React.PropTypes.bool,
  text: React.PropTypes.string,
};

export default Markdown;
