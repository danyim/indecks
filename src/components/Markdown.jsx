import React from 'react';
import Remarkable from 'remarkable';

const propTypes = {
  className: React.PropTypes.string,
  removeParaTags: React.PropTypes.bool,
  text: React.PropTypes.string,
  handleOnClick: React.PropTypes.func
};

const defaultProps = {
  className: '',
  text: '',
  removeParaTags: false,
  handleOnClick: null,
};

class Markdown extends React.Component {
  convertToMarkdown(plaintext = '', removeParaTags = false) {
    const md = new Remarkable();
    let rawMarkup = md.render(plaintext);
    if (removeParaTags === true) {
      rawMarkup = rawMarkup.replace('<p>', '');
      rawMarkup = rawMarkup.replace('</p>', '');
    }
    return { __html: rawMarkup };
  }

  render() {
    return (
      <span
        className={this.props.className}
        onClick={this.props.handleOnClick}
        dangerouslySetInnerHTML={
          this.convertToMarkdown(this.props.text, this.props.removeParaTags)
        }
      />
    );
  }
}

Markdown.propTypes = propTypes;
Markdown.defaultProps = defaultProps;

export default Markdown;
