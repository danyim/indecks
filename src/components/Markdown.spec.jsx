import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Markdown from './Markdown'

const defaultProps = {
  className: 'class',
  text: 'Text',
  handleOnClick: jest.fn()
}

function setup(props = defaultProps) {
  const wrapper = shallow(<Markdown {...props} />)

  return {
    props,
    wrapper
  }
}

describe('Markdown', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup()
    expect(wrapper.find('div').hasClass(defaultProps.className)).toBe(true)

    const tree = renderer
      .create(<Markdown text="Test">Hello</Markdown>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call the handleOnClick once when Markdown is clicked', () => {
    const handler = jest.fn()
    const { wrapper } = setup({
      ...defaultProps,
      handleOnClick: handler
    })

    wrapper.find('div').simulate('click')
    expect(handler.mock.calls.length).toBe(1)
  })

  // it('should convert text into Markdown', () => {
  //   const remarkable = jest.mock('Remarkable');
  //   const text = 'Test text string';
  //   const { wrapper } = setup({
  //     ...defaultProps,
  //     text: `# ${text}`
  //   });

  //   expect(wrapper.find('h1').text()).toEqual(text);
  // });
})
