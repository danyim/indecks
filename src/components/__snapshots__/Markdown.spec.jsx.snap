// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown should render self and subcomponents 1`] = `
<div
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Test</p>
",
    }
  }
  onClick={null}
/>
`
